scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1038,22 +1039,25 @@
   "Autoscales the given pool to satisfy the given pending jobs, if:
   - There is at least one pending job
   - There is at least one compute cluster configured to do autoscaling"
-  [pending-jobs pool-name compute-clusters job->acceptable-compute-clusters-fn]
+  [pending-jobs-for-autoscaling pool-name compute-clusters job->acceptable-compute-clusters-fn]
   (timers/time!
     (timers/timer (metric-title "trigger-autoscaling!-duration" pool-name))
     (try
       (let [autoscaling-compute-clusters (filter #(cc/autoscaling? % pool-name) compute-clusters)
             num-autoscaling-compute-clusters (count autoscaling-compute-clusters)]
-        (when (and (pos? num-autoscaling-compute-clusters) (seq pending-jobs))
-          (let [{:keys [max-jobs-for-autoscaling]} (config/kubernetes)
-                pending-jobs-for-autoscaling (take max-jobs-for-autoscaling pending-jobs)
-                compute-cluster->jobs (distribute-jobs-to-compute-clusters
+        (when (and (pos? num-autoscaling-compute-clusters) (seq pending-jobs-for-autoscaling))
+          (log/info "In" pool-name "pool, preparing for autoscaling")
+          (let [compute-cluster->jobs (distribute-jobs-to-compute-clusters
                                         pending-jobs-for-autoscaling pool-name autoscaling-compute-clusters
                                         job->acceptable-compute-clusters-fn)]
             (log/info "In" pool-name "pool, starting autoscaling")
-            (doseq [[compute-cluster jobs-for-cluster] compute-cluster->jobs]
-              (cc/autoscale! compute-cluster pool-name jobs-for-cluster adjust-job-resources-for-pool-fn))
-            (log/info "In" pool-name "pool, done autoscaling"))))
+            (->> compute-cluster->jobs
+                 (map
+                   (fn [[compute-cluster jobs-for-cluster]]
+                     (future (cc/autoscale! compute-cluster pool-name jobs-for-cluster adjust-job-resources-for-pool-fn))))
+                 doall
+                 (run! deref)))
+          (log/info "In" pool-name "pool, done autoscaling")))
```

Comment:
Indentation seems off on this line

Reply:
Indentation is correct (at least per cursive's default rules) in this section.